### PR TITLE
Refactor addBackendPolicyIfConfigured to allow future features

### PR DESCRIPTION
**Pull request by sawsa307**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Refactor IR in GCE to allow additional features.

**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: pkg/i2gw/providers/gce/extensions/output_extensions.go
Comment (spencerhance) [inline pkg/i2gw/providers/gce/extensions/output_extensions.go:30]: Nit: how do we usually handle invalid config?  Could we add a log here if the backendConfig had `affinityCookieTtlSec` set without `affinityType` of `GENERATED_COOKIE`?

**Review by spencerhance — APPROVED** [review]
Thanks @sawsa307!

**Comment by spencerhance** [comment]
official /lgtm /approve 

(although I don't have OWNERS approval)

File: pkg/i2gw/providers/gce/extensions/output_extensions.go
Comment (sawsa307) [reply pkg/i2gw/providers/gce/extensions/output_extensions.go:30]: Sounds good! Created a function to check backend config before processing translation.

**Comment by robscott** [comment]
Thanks @sawsa307! Will defer to @LiorLieberman for final approval.

/lgtm

**Comment by LiorLieberman** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**
